approximate_moran.py

```diff
@@ -0,0 +1,59 @@
+import axelrod as axl
+import numpy as np
+
+class Pdf:
+    """A class for a probability distribution"""
+    def __init__(self, counter):
+        """Take as an instance of collections.counter"""
+        self.sample_space, self.counts = zip(*counter.items())
+        self.size = len(self.sample_space)
+        self.total = sum(self.counts)
+        self.probability = list([v / self.total  for v in self.counts])
+
+    def sample(self):
+        """Sample from the pdf"""
```

Comment:
This is fitness proportionate selection, essentially. The fastest way to do this is to compute the normalized cumulative sums and then use a random sample on [0, 1] like so:
```
csums = np.cumsum(normalize([v for k,v in items]))
r = random.random()
for j, x in enumerate(csums):
    if x >= r:
return j
```

You can also precache the cumulative sums. This is what my [mpsim](https://github.com/marcharper/mpsim/blob/master/mpsim/simulation_.py) library does to simulate trajectories for Markov processes.